tests/cli/test_train_model.py

```diff
@@ -963,3 +963,23 @@ def test_train_wandb_logger(monkeypatch, tmp_path):
 
     assert "'base_precision': 64" in file_log
     assert "'seed': 42" in file_log
+
+
+@pytest.mark.parametrize(
+    "value, expected",
+    [
+        (0, "0"),
+        (999, "999"),
+        (1000, "1K"),
+        (1234, "1.2K"),
+        (1254, "1.3K"),
+        (20454, "20.5K"),
+        (306945, "307K"),
+        (1000000, "1M"),
+        (1500000000, "1.5B"),
+        (1865039582492, "1.9T"),
+        (1835039582492, "1.8T"),
+    ],
+)
+def test_human_readable_parameter_counter(value, expected):
+    assert _human_readable(value) == expected
```

Comment:
Sorry but we don't test private functions. Can you apply this to the public function where `_human_readable` is used.

Reply:
> Or make human_readable public...

Why should this be a public function? Do you need this when developing an architecture?

> I don't see the point in not unit testing a function, even if it is private 😅

Because if it is private function you can just change the API or remove it, without consequences. You design a library API against a public tested API and not private ones. This is style of [software project organization](https://stackoverflow.com/questions/105007/should-i-test-private-methods-or-only-public-ones) that we follow in the lab. We can of course drop this concept but then we should start a bigger discussion about it.

Reply:
I was just proposing to make it public so that we can test it without conflicting with this rule. In this case, unit testing the function is much simpler than testing all the cases through the CLI.

The link you shared refers specifically to private methods, which I think it is different since they are attached to an instance and might not follow functional programming. Testing a functional function whether it is private or public I think it's a good idea, specially if testing it through the public interface involves a lot of complexity. But ok, I understand that you don't want to change this now, that's why I proposed to make it public.

Reply:
> Do you need this when developing an architecture?

Why not, who knows haha